### PR TITLE
RandomX

### PR DESCRIPTION
**Pull request by jagerman**
This implements a switch to RandomX (custom Loki variant) hashing at the v12 fork.

### CODE REVIEW

**Comment by jagerman** [comment]
Not quite ready to merge -- need to add an appropriate difficulty override for mainnet for a few blocks after the fork height.

File: src/rpc/core_rpc_server.cpp [hunk 1]
```diff
@@ -940,6 +940,8 @@ namespace cryptonote
     res.address = get_account_address_as_str(nettype(), false, lMiningAdr);
     const uint8_t major_version = m_core.get_blockchain_storage().get_current_hard_fork_version();
 
+    if (major_version >= network_version_12_checkpointing)
+      res.pow_algorithm = "RandomX (LOKI variant)";
     if (major_version >= network_version_7 && major_version <= network_version_10_bulletproofs)
```
Comment (Doy-lee) [inline src/rpc/core_rpc_server.cpp]: else if

Comment (jagerman) [reply src/rpc/core_rpc_server.cpp]: Fixed (and replaced the whole thing with cleaner looking ternary operators).

Comment (tewinget) [reply src/rpc/core_rpc_server.cpp]: hmm...testing my memory here, but did we start on CN Heavy?  I feel like we changed to that pretty early but can't recall if we started there.

**Pull request merged**